packages/flutter_tools/lib/src/build_system/file_hash_store.dart

```diff
@@ -143,24 +137,27 @@ class FileHashStore {
 
   /// Computes a hash of the provided files and returns a list of entities
   /// that were dirty.
-  // TODO(jonahwilliams): compare hash performance with md5 tool on macOS and
-  // linux and certutil on Windows, as well as dividing up computation across
-  // isolates. This also related to the current performance issue with checking
-  // APKs before installing them on device.
   Future<List<File>> hashFiles(List<File> files) async {
     final List<File> dirty = <File>[];
-    for (File file in files) {
-      final String absolutePath = file.resolveSymbolicLinksSync();
-      final String previousHash = previousHashes[absolutePath];
-      final List<int> bytes = file.readAsBytesSync();
-      final String currentHash = md5.convert(bytes).toString();
+    await Future.wait(<Future<void>>[
+      for (File file in files) _hashFile(file, dirty, Pool(32))]);
```

Comment:
How was the limit of 32 chosen?

Reply:
Its a nice sounding number. There's no real way to know though

Reply:
Ah. I'd just leave a comment that this number is arbitrary. Otherwise, it looks purposeful. Someone may be scared to change it later.

Reply:
Did you one better and moved it to a constant - also tried making this async